include/openssl/core_names.h

```diff
@@ -128,6 +130,16 @@ extern "C" {
 #define OSSL_KDF_PARAM_SSHKDF_TYPE  "type"      /* int */
 #define OSSL_KDF_PARAM_SIZE         "size"      /* size_t */
 
+/* Known KDF names */
+#define OSSL_KDF_NAME_HKDF          SN_hkdf
+#define OSSL_KDF_NAME_PBKDF2        LN_id_pbkdf2
+#define OSSL_KDF_NAME_SCRYPT        SN_id_scrypt
+#define OSSL_KDF_NAME_SSHKDF        SN_sshkdf
+#define OSSL_KDF_NAME_SSKDF         SN_sskdf
+#define OSSL_KDF_NAME_TLS1_PRF      SN_tls1_prf
+#define OSSL_KDF_NAME_X942KDF       SN_x942kdf
+#define OSSL_KDF_NAME_X963KDF       SN_x963kdf
```

Comment:
I would rather have the explicit strings here

Reply:
Changed.

I'm not keen on introducing unnecessary redundancy but not sufficiently against it to bother.